x-pack/filebeat/input/entityanalytics/provider/okta/internal/okta/okta.go

```diff
@@ -164,12 +214,81 @@ func (o Response) String() string {
 func GetUserDetails(ctx context.Context, cli *http.Client, host, key, user string, query url.Values, omit Response, lim *rate.Limiter, window time.Duration) ([]User, http.Header, error) {
 	const endpoint = "/api/v1/users"
 
-	u := url.URL{
+	u := &url.URL{
 		Scheme:   "https",
 		Host:     host,
 		Path:     path.Join(endpoint, user),
 		RawQuery: query.Encode(),
 	}
+	return getDetails[User](ctx, cli, u, key, user == "", omit, lim, window)
+}
+
+// GetDeviceDetails returns Okta device details using the list devices API endpoint. host is the
+// Okta user domain and key is the API token to use for the query. If device is not empty,
+// details for the specific device are returned, otherwise a list of all devices is returned.
+//
+// See GetUserDetails for details of the query and rate limit parameters.
+//
+// See https://developer.okta.com/docs/api/openapi/okta-management/management/tag/Device/#tag/Device/operation/listDevices for details.
+func GetDeviceDetails(ctx context.Context, cli *http.Client, host, key, device string, query url.Values, lim *rate.Limiter, window time.Duration) ([]Device, http.Header, error) {
```

Comment:
Small nit - can we optimise this function signature? Seems a bit long.

Reply:
This seems clear to me; at a call site it lives on a single line within a screen width.